### ISSUE TITLE
Eager setting in request should always override setting of the driver

### ISSUE THREAD

**Issue by arnodenuijl** [cyclejs/cycle-http-driver#34]
At this moment you can set ```eager``` when you create the driver and you can set ```eager``` in the request. 
  When ```eager``` is set to ```false``` in the driver you can force it to ```true``` in the request. 
  But when ```eager``` is set to ```true``` in the driver you cannot override it to ```false``` in the request. 
  
  I would expect explicitely setting ```eager``` to either ```true``` or ```false``` in the request to always override 
  whatever the setting is in the driver. Is this assumption correct? If so, i would like to create a pull request to 
  fix this. Might need some help because this would by me first pull request for an open source project. 
  
  So the resulting behavior from all the possible settings would be as follows:
  
  Eager setting driver | Eager setting request | result
  -------------- | --------------- | ------
  not explicitly set | not explicitly set | not eager *(default of the driver)*
  not explicitly set | true | eager
  not explicitly set | false | not eager
  true | not explicitly set | eager *(driver setting)*
  true | true | eager
  true | false | not eager *(request setting always overrides driver setting)*
  false | not explicitly set | not eager *(driver setting)*
  false | true | eager *(request setting always overrides driver setting)*
  false | false | not eager 
 
Unit test to show behavior:
```
  it('should not auto-execute HTTP request when factory gets eager = true but the requests explicitly overrides with eager = false',
    function(done) {
      var request$ = Rx.Observable.just({
        url: uri + '/pet',
        method: 'POST',
        send: {name: 'Woof', species: 'Dog'},
        eager: false
      });
      var httpDriver = makeHTTPDriver({eager: true});
      globalSandbox.petPOSTResponse = null;
      httpDriver(request$);
      setTimeout(function () {
        assert.strictEqual(globalSandbox.petPOSTResponse, null);
        done();
      }, 100);
    }
  );
```

**Comment by whitecolor** [comment]
I think this a correct proposal, you could make a PR. If `request.eager` option present this should override driver's option anyway.

**Comment by TylorS** [comment]
@arnodenuijl Thank you for the very nice write up. I believe that would be the correct and most expected behavior. A pull request would be greatly appreciated. 

Since you mention this being your first OSS project submission, first and foremost thank you for choosing Cycle.js :+1: 

All we will generally ask for

  1. The fix/feature, of course
  2. Tests supporting the fix/feature, as applicable
  3. Adding documentation, where and as applicable
  4. Descriptive commit messages (why? how?) so when we look back we know what happened/changed and for what

**Comment by arnodenuijl** [comment]
Cool cool cool. Coming soon :-)

* Resulting changes in the dist folder should not be included?
* ```npm run test-browser``` borks out 
  ```
> @cycle/http@7.0.0 test-browser /home/arno/Projects/cycle-http-driver

> zuul --open --no-coverage --local 4000 -- test/browser.js

Error: ENOENT: no such file or directory, lstat '/home/arno/Projects/cycle-http-driver/dist/cycle-http-   driver'
    at Error (native)
```
  but it also does that with a clean copy. I understand he can't find the file, it's missing the extension. Is this broken with you guys to? Or is this a local thing?

**Comment by arnodenuijl** [comment]
Created a pull request. Shoot away!

**Comment by whitecolor** [comment]
Not very evident)
```
reqOptions.eager || eager && reqOptions.eager !== false
```
Would make something like this:
```
reqOptions.eager !== undefined ? eager : reqOptions.eager
```
And probably no need to modify `dist..

**Comment by arnodenuijl** [comment]
I see where that is not very evident. 

It first had parentheses which made it somewhat clearer.
 
```
reqOptions.eager || (eager && reqOptions.eager !== false)
```

But ESLint punished me with a ```Gratuitous parentheses around expression``` error

What you propose:
```
reqOptions.eager !== undefined ? reqOptions.eager : eager
```

is slayed by ESLint with ```Unexpected use of undefined``` error 

And that can be fixed with 

```
typeof reqOptions.eager !== `undefined` ? reqOptions.eager : eager
```
But that gives the warning ```Line 131 exceeds the maximum line length of 80```

We can fix that by separating the type check
```
let eagerDefinedOnRequest = typeof reqOptions.eager !== `undefined`
if (eagerDefinedOnRequest ? reqOptions.eager : eager) {
```

I can live with the last two both. What's your preference? Or any alternative.

**Comment by whitecolor** [comment]
probably
```
reqOptions.hasOwnProperty('eager')
  ? reqOptions.eager : eager
```

**Comment by arnodenuijl** [comment]
Check. Didn't know that one. Updated the pull request, ditched the changes in the dist folder.

**Comment by TylorS** [comment]
@arnodenuijl You can simply do
```js
typeof reqOptions.eager !== `undefined` ?
  reqOptions.eager : eager
```
Feel free to split things onto multiple lines. It makes things much more clear. Appreciate that you're trying to keep our ESLint configuration happy, it's quite strict.

@whitecolor `npm run dist` to create a dist/ build. But only @staltz can merge this, and do the releases.

**Comment by staltz** [comment]
@arnodenuijl Very good initiative! Thanks for the thorough analysis, and I agree.

**Comment by staltz** [comment]
Done and released in v8.0.0.
